### PR TITLE
PSR-1380 Fixed test coverage

### PR DESCRIPTION
**Pull request by westwater**
- Updated coverage to around 95% 
- added "sbt testc" as an easier way to run coverage
- I had to disable certain files when running coverage against unit tests as they are covered by the it:tests (because of how we have inetgration tests as a seperate sbt module, i'm not sure how to run them as part of the core coverage run)

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -46,3 +54,8 @@ lazy val it = project
     Test / scalafmtOnCompile := true,
     Test / unmanagedResourceDirectories += baseDirectory.value / "it" / "test" / "resources"
   )
+  .settings(inConfig(Test)(testSettings) *)
+  .settings(resolvers += Resolver.jcenterRepo)
+  .settings(CodeCoverageSettings.settings *)
+
+addCommandAlias("testc", "; clean ; coverage ; test ; coverageReport ;")
```
Comment (westwater) [inline build.sbt:61]: I tried to add it:tests here to include them in the coverage run which requires switching sbt projects part way through but unfortunately it doesn't persist the coverage window - if anyone knows how to now run it:tests with our unit tests for coverage please let me know

Comment (cyeniceri) [reply build.sbt:60]: `  .settings(inConfig(Test)(testSettings) *)`
do we need this line ? 49-54 are same. Should we use 49-54 only or reuse testSettings only ?

Comment (cyeniceri) [reply build.sbt:60]: addCommandAlias("testc", "; clean ; coverage ; test ; it:test ; coverageReport")
Is this not working as expected ?

Comment (cyeniceri) [reply build.sbt:60]: We have already a file (runtests.sh) under the root directory. Is this not helping us enough to see the coverage ?

Comment (westwater) [reply build.sbt]: I'm not a big fan of using that file because we can't tailor the sbt run settings on the fly (more sbt process memory for instance)

**Review by cyeniceri — APPROVED** [review]

**Pull request merged**